File: solutions/r/secret-handshake/1/secret-handshake.R
library(R.utils)

handshake <- function(n) {
  binS = unlist(strsplit(intToBin(n), split = ""))
  len = length(binS)
  res = c()
  if(len == 1){
    if(binS == "1"){
      return (c("wink"))
    }
  }
  for (i in seq(len,1)) {
    if (binS[i:i] == "1") {
      switch(len-i+1,
       "1" = {res = c(res,"wink")},
       "2" = {res = c(res,"double blink")},
       "3" = {res = c(res,"close your eyes")},
       "4" = {res = c(res,"jump")},
       "5" = {res = rev(res)}
             )
    }
  }
  return(res)
}

File: solutions/python/secret-handshake/1/secret_handshake.py
commands_list = ['wink', 'double blink', 'close your eyes', 'jump', 'reverse']


def commands(binary_str: str) -> list[str]:
    cmd_list = [commands_list[i] for i, bit in enumerate(list(binary_str)[::-1]) if bit == '1']
    if 'reverse' in cmd_list:
        cmd_list.remove('reverse')
        cmd_list.reverse()
    return cmd_list
